src/parser/mod.rs

```diff
@@ -9225,8 +9225,63 @@ impl<'a> Parser<'a> {
     }
 
     pub fn parse_use(&mut self) -> Result<Statement, ParserError> {
-        let db_name = self.parse_identifier(false)?;
-        Ok(Statement::Use { db_name })
+        // What should be treated as keyword in given dialect
+        let allowed_keywords = if dialect_of!(self is HiveDialect) {
```

Comment:
It'd be better to include `GenericDialect` as well. cc @iffyio

Reply:
AFAIU it does not have any specific keywords, so right now it would fall into `else` clause but you'd like to have a separate `if` for it with empty keywords?

Reply:
@kacpermuda oh, sorry for not stating this clearly. What I mean is to include `GenericDialect` inside `dialect_of!`, so that the new syntaxes are valid to the generic dialect. For example;

`let allowed_keywords = if dialect_of!(self is HiveDialect)` can be changed to `let allowed_keywords = if dialect_of!(self is HiveDialect | GenericDialect)`

Reply:
Okay, I see. Maybe i should create a separate if for the `GenericDialect` that will include all the keywords possible in all other dialects? Otherwise when using `GenericDialect` we will fall into whatever `if` is first. Not sure how this `GenericDialect` works 😄

Reply:
@kacpermuda My bad, I forgot the if condition only contains dialect. It makes sense to add another one for GenericDialect. Or just keep the current implementation and to see if @iffyio has any comments.

Reply:
Sure, let's wait and see then.

Reply:
Ah since support varies quite a bit, we could have GenericDialect preserve the old behavior supporting just `USE abc`? thinking since that seems to be the most common variant

Reply:
Yes, i think with current behaviour that's exactly what's happening - the `GenericDialect` falls into else clause with no keywords, thus supporting only `USE mydb;`. Is that okay for you or should i still create another `else if` for `GenericDialect` just so it's explicit ?

Reply:
Sounds good! We can keep it as is in that case yeah